gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -889,8 +802,20 @@ private static void checkPartitionCompatibility(Partition desiredTargetPartition
         referenceAuthority = uri.getAuthority();
       }
 
+      if (!this.dataset.getTableRootPath().isPresent()) {
```

Comment:
The big performance issue is the `getFileStatus` call, but that call is cached. We might want to optimize in the future, for now, this is how things have been running so far (this was just refactoring), so I don't have big concerns with performance.